networkx/algorithms/decomposition.py

```diff
@@ -0,0 +1,174 @@
+"""
+Find the decompositions of a graph.
+
+Two techniques are implemented: k-core and k-peak.
+
+The highest possible k (for core or peak) is called degeneracy.
+
+See the following references for details:
+
+Graph core: https://networkx.org/documentation/stable/reference/algorithms/core.html
+
+The k-peak Decomposition: Mapping the Global Structure of Graphs, Govindan Priya, Wang Chenghong, Xu Chumeng, Duan Hongyu, and Soundarajan Sucheta, WWW '17. https://doi.org/10.1145/3038912.3052635
+
+"""
+
+import networkx as nx
+
+__all__ = [
+    "k_core_decomposition",
+    "k_core_degeneracy",
+    "k_peak_decomposition",
+    "k_peak_degeneracy",
+]
+
+
+@nx.utils.not_implemented_for("multigraph")
+def k_core_decomposition(G, k, core_number=None):
```

Comment:
I'm -1 on adding wrapper functions with different names. Doing so just makes the API harder to maintain and understand.

Reply:
> it may become required if additional graph decomposition methods are introduced/added in the future. It's important to note that graph decomposition is a broader concept than k-core.

In my experience, prematurely engineering something for what *might* be implemented leads to poor API design. I'd vote to address proposals for new algorithms on a case-by-case basis!